sklearn/utils/estimator_checks.py

```diff
@@ -1443,41 +1443,87 @@ def check_supervised_y_2d(name, estimator_orig):
     assert_allclose(y_pred.ravel(), y_pred_2d.ravel())
 
 
-@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+@ignore_warnings
+def check_classifiers_classes_helper(X, y, name, classifier_orig):
+    classes = np.unique(y)
+    classifier = clone(classifier_orig)
+    if name == 'BernoulliNB':
+        X = X > X.mean()
+    set_random_state(classifier)
+    # fit
+    classifier.fit(X, y)
+
+    y_pred = classifier.predict(X)
+
+    if hasattr(classifier, "decision_function"):
+        decision = classifier.decision_function(X)
+        n_samples, n_features = X.shape
+        msg_shape="Predicted shape of `decision_function` mismatches the expected " \
+                  "shape: expected '%s', got '%s'"
+        if len(classes) == 2:
+            assert_equal(decision.shape, (n_samples, ), msg_shape % \
```

Comment:
For future reference, we no longer need to use helpers like this, and can use a bare assert instead.

Reply:
I'm not sure we need to test shape separate to content.

Reply:
Shape can be seen as a precheck prior to content. I can remove it if necessary.

Reply:
assert_array_equal will do this pre-check